deerc-dev/openedx-admin
# Simple server

Created simple server with the single endpoint. When reaching `localhost:8080/hello`, it returns message `Hello!` with the status code `200 OK`.

How to reproduce:
1. Install `cargo`, the Rust package manager. Preferably use [`rustup`](https://rustup.rs/).
2. Launch it via `cargo run` in `backend` directory.
3. Reach `localhost:8080/hello`, the server should return `Hello!`.

Additionaly, `SERVER_HOST` environmental variable can be set with specific host and port. For example, `127.0.0.1:1488`.

When dockerizing, remember that the Rust ecosystem has massive memory size (up to 1.5 Gbytes) and the Rust compiler produces heavy binaries for dependencies and the program itself. Don't forget to add `target` to `.dockerignore`. Additionaly, I recommend to skim [related articles](https://dev.to/sergeyzenchenko/actix-web-in-docker-how-to-build-small-and-secure-images-2mjd).
